### PR TITLE
[PLAT-1054] Unblock final certificates

### PR DESCRIPTION
**Pull request by preziu**
Users that have finished course after 15/02 can not download final certificate - but it could be possible fim them to do so until the end of course access.

### CODE REVIEW

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php [hunk 1]
```diff
@@ -117,13 +117,6 @@ public function getFinalCertificate($orderId)
 			return $this->respondForbidden("User not allowed to view order details");
 		}
 
-		if (!$user->hasFinishedCourse(
```
Comment (kvas-damian) [inline app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: Why do we remove this condition? Don't we need to validate if the user finished the course before we generate a certificate?

File: resources/assets/js/components/user/UserCertificates.vue [hunk 2]
```diff
@@ -25,7 +25,7 @@
 			<ul>
 				<li v-for="order in finalCertificates" :key="order.id">
 					<a @click="downloadFinalCertificate(order.id)">
-						Certyfikat Ukończenia: {{order.product.name}} - {{formatDate(order.product.course_start)}} - {{formatDate(order.product.course_end)}}
+						Certyfikat Ukończenia: {{order.product.name}} - {{formatDate(order.product.course_start)}} - {{formatDate(new Date())}}
```
Comment (kvas-damian) [inline resources/assets/js/components/user/UserCertificates.vue]: Are you sure we want to show the current date for courses that finished a year ago?

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php [hunk 3]
```diff
@@ -136,7 +129,7 @@ public function getFinalCertificate($orderId)
 		$img->text(
 			sprintf("%s - %s r.",
 				$order->product->course_start->format('j.m.Y'),
-				$order->product->course_end->format('j.m.Y')
+				$order->product->access_end->format('j.m.Y')
```
Comment (kvas-damian) [inline app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: I don't think it's ok if we sign a certificate with future date.

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php [hunk 4]
```diff
@@ -150,7 +143,7 @@ public function getFinalCertificate($orderId)
 			$font->align('center');
 		});
 
-			$img->text($order->product->course_end->format('j / m / Y') . ' r., Poznań', 1182, 867, function($font) {
+			$img->text($order->product->access_end->format('j / m / Y') . ' r., Poznań', 1182, 867, function($font) {
```
Comment (kvas-damian) [inline app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: I don't think it's ok that we sign a certificate with a future date.

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php [hunk 5]
```diff
@@ -87,7 +87,7 @@ public function getParticipationCertificate($orderId)
 			$font->align('center');
 		});
 
-		$img->text($order->product->course_start->format('j / m / Y') . ' r., Poznań', 2365, 1650, function($font) {
+		$img->text(Carbon::now()->format('j / m / Y') . ' r., Poznań', 2365, 1650, function($font) {
```
Comment (kvas-damian) [inline app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: The date on the certificate will be different on every request. I'd be confused when I get it with different data.

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php (hunk 1 above)
Comment (preziu) [reply app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: The idea behind letting a user download a final certificate is that he can do it after he meets a certain criteria during the course. There is no specific date to be achieved, only course progress conditions are valid. For example: a user met the final criteria at 23.02, but his actual access to the course ends at 1.05. So when we restrict users to the course_end, which at the moment is 15.02, we are closing the opportunity for them to get the final certificate :)

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php (hunk 5 above)
Comment (preziu) [reply app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: My bad! this should stay as a `course_start`, I'll push a fix right away 👍

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php (hunk 4 above)
Comment (preziu) [reply app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: It a certain type of promise. The desired behavior is that we put a date of actual meeting the criteria. This is only a quick fix until we include a proper logic for calculating the actual date.

File: resources/assets/js/components/user/UserCertificates.vue (hunk 2 above)
Comment (preziu) [reply resources/assets/js/components/user/UserCertificates.vue]: Well, I couldn't really think about a proper date to set the finishing date, but after a thought i think it should be a `access_end`. I'm pushing a fix right away!

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php (hunk 1 above)
Comment (kvas-damian) [reply app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: But right now we don't check if "course progress conditions are valid". Now, everyone can hit this endpoint and generate a certificate for any of his orders - without any checks.

File: app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php (hunk 4 above)
Comment (kvas-damian) [reply app/Http/Controllers/Api/PrivateApi/CertificatesApiController.php]: Please create a ticket and mark this line with TODO comment.

**Pull request merged**